synapse/handlers/federation_event.py

```diff
@@ -715,7 +722,7 @@ async def _get_missing_events_for_pdu(
 
     @trace
     async def _process_pulled_events(
-        self, origin: str, events: Iterable[EventBase], backfilled: bool
+        self, origin: str, events: List[EventBase], backfilled: bool
```

Comment:
Since we're iterating over this multiple times, made sure we're passing this a `List` now

Reply:
We usually use `Collection` for this because it's read-only:
```suggestion
        self, origin: str, events: Collection[EventBase], backfilled: bool
```

`Sequence` would work too, but we don't really need a list-like collection here and it rules out things like sets.